src/error/fowler-noise.lisp

```diff
@@ -0,0 +1,190 @@
+  (make-instance 'fowler-stabilizer-qvm
+                 :noise-class (fowler-qvm-noise-class qvm)
+                 :noise-probability (fowler-qvm-noise-probability qvm)
+                 :tableau (alexandria:copy-array (qvm::stabilizer-qvm-tableau qvm))
+                 :gate-definitions (qvm::gate-definitions qvm)
+                 :wait-function (qvm::wait-function qvm)
+                 :classical-memory-subsystem (qvm::classical-memory-subsystem qvm)))
+
+(defmethod compile-loaded-program ((qvm fowler-qvm))
+  qvm)
+
+;;;
+;;; read individual flags out of the error-type bitvector
+;;;
+
+(defmacro fowler-qvm-noise-identity? (qvm)
+  `(logbitp 0 (fowler-qvm-noise-class ,qvm)))
+
+(defmacro fowler-qvm-noise-1Q-gate? (qvm)
+  `(logbitp 1 (fowler-qvm-noise-class ,qvm)))
+
+(defmacro fowler-qvm-noise-2Q-gate? (qvm)
+  `(logbitp 2 (fowler-qvm-noise-class ,qvm)))
+
+(defmacro fowler-qvm-noise-reset? (qvm)
+  `(logbitp 3 (fowler-qvm-noise-class ,qvm)))
+
+(defmacro fowler-qvm-noise-readout? (qvm)
+  `(logbitp 4 (fowler-qvm-noise-class ,qvm)))
+
+;;;
+;;; basic qvm behavior definitions
+;;;
+
+(defmethod %fast-apply ((qvm fowler-pure-state-qvm) instr-name qubit)
+  (let* ((gate (quil:gate-definition-to-gate (quil:lookup-standard-gate instr-name)))
+         (pure-state (qvm::state qvm)))
+    (apply-gate-to-state gate pure-state (list (quil:qubit qubit)))))
+
+(defmethod %fast-apply ((qvm fowler-stabilizer-qvm) instr-name qubit)
+  (let* ((instr (quil::build-gate instr-name () qubit))
+         (clifford (qvm::gate-application-to-clifford instr)))
+    (apply (qvm::compile-clifford clifford)
+           (qvm::stabilizer-qvm-tableau qvm)
+           (list qubit))))
+
+;;;
+;;; noise behavior definitions
+;;;
+
+(defmethod transition :after ((qvm fowler-qvm) (instr cl-quil:application))
+  (case (length (cl-quil::application-arguments instr))
+    (1
+     (cond
+       ((equalp (quil::named-operator "I")
+                (cl-quil::application-operator instr))
+        (when (fowler-qvm-noise-identity? qvm)
+          (let ((qubit (quil:qubit-index (first (cl-quil::application-arguments instr))))
+                (p/3 (/ (fowler-qvm-noise-probability qvm) 3)))
+            (multiprobabilistically
+              (p/3 (%fast-apply qvm "X" qubit))
+              (p/3 (%fast-apply qvm "Y" qubit))
+              (p/3 (%fast-apply qvm "Z" qubit))))))
+       ((equalp (quil::named-operator "H")
+                (cl-quil::application-operator instr))
+        (when (fowler-qvm-noise-1Q-gate? qvm)
+          (let ((qubit (quil:qubit-index (first (cl-quil::application-arguments instr))))
+                (p/3 (/ (fowler-qvm-noise-probability qvm) 3)))
+            (multiprobabilistically
+              (p/3 (%fast-apply qvm "X" qubit))
+              (p/3 (%fast-apply qvm "Y" qubit))
+              (p/3 (%fast-apply qvm "Z" qubit))))))
+       (t
+        (break)
```

Comment:
I think the `break`s should be taken out, and replaced with a continuable error with `cerror`